tf2_ros/src/buffer.cpp

```diff
@@ -190,6 +191,73 @@ Buffer::canTransform(const std::string& target_frame, const tf2::TimePoint& targ
   return retval; 
 }
 
+TransformStampedFuture
+Buffer::waitForTransform(const std::string& target_frame, const std::string& source_frame, const tf2::TimePoint& time,
+                         const tf2::Duration& timeout, TransformReadyCallback callback)
+{
+  if (nullptr == timer_interface_) {
+    throw std::runtime_error("timer interface not set before call to waitForTransform");
```

Comment:
This should be a custom Exception class. At least a generic tf2 Exception.

Reply:
Thanks for the review! I've added a custom exception in 7556ee5